src/Text/Markdown/SlamDown/Html.purs

```diff
@@ -215,6 +217,13 @@ evalSlamDownQuery e =
     GetFormState k -> do
       SlamDownState state <- H.get
       pure $ k state.formState
+    PopulateForm values next -> do
+      SlamDownState { document } <- H.get
+      let desc = formDescFromDocument document
+          keysToPrune = filter (`M.member` desc) (toList (M.keys values))
+          prunedFormState = foldr M.delete values keysToPrune
```

Comment:
I'm a little confused by the logic here. It looks like it's finding out which keys of `values` are already part of the document, and then removing them from `values` prior to updating the state. Is this what we want to do, or have I got turned around somehow?